### ISSUE TITLE
Decompiler's output for structures corrupts after `typedef`ing it

### ISSUE THREAD

**Issue by kevre91** [Vector35/binaryninja-api#3304]
**Version and Platform (required):**
- Binary Ninja Version: 3.1.3469 
- OS: Windows
- OS Version: 10
- CPU Architecture: x64

I've a structure named `list_T` like

```c
struct __packed list_T
{
    void** items;
    uint32_t size;
    int32_t item_size;
};
``` 

and a function named `init_list` which have this HLIL output:

```c
00551000  struct list_T* init_list(int32_t item_size)

0055100f      struct list_T* list = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, 12)
0055101b      list->size = 0
00551028      list->item_size = item_size
0055102e      list->items = 0
00551039      OutputDebugStringA(&data_553000)
00551045      return list
```

I don't like `struct` keyword in output, so I replaced my `list_T` structure with `typedef` for `list_T` in order to get rid of `struct` keyword like this:

```c
typedef struct __packed _list_T
{
    void** items;
    uint32_t size;
    int32_t item_size;
}list_T;
```

But, this modification corrupts decompiler output:

```c
00551000  list_T* init_list(int32_t item_size)

0055100f      list_T* list = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, 12)
0055101b      list->__offset(0x4).d = 0
00551028      list->__offset(0x8).d = item_size
0055102e      list->__offset(0x0).d = 0
00551039      OutputDebugStringA(&data_553000)
00551045      return list
```

Here is a screenshot of Types after replacing `list_T` structure:

![structure](https://user-images.githubusercontent.com/109617473/180050270-cb34c6f7-7199-419a-88ab-ecc9e67ea718.PNG)

So, the structure and new type definition is exist for sure.

I also tested it with directly creating a structure in a fresh BN session like this:

```c
typedef struct __packed _list_T
{
    void** items;
    uint32_t size;
    int32_t item_size;
}list_T;
```

But it still outputs same results. By the way, If I change type of `list` variable to `struct _list_T*` it works again.

Is this a bug?

**Comment by psifertex** [comment]
What version of BN are you using? We believe this is already fixed on the latest dev.

**Comment by CouleeApps** [comment]
Yeah, this is fixed as of 3.1.3499-dev
<img width="639" alt="image" src="https://user-images.githubusercontent.com/711973/180052862-8a6ff8a8-67fa-49e4-81fe-c4e43b76c106.png">
